src/segment-loader.js

```diff
@@ -992,16 +992,16 @@ export default class SegmentLoader extends videojs.EventTarget {
 
     log('handleUpdateEnd_');
 
-    if (!segmentInfo.isSyncRequest) {
-      this.mediaIndex = segmentInfo.mediaIndex;
-      this.fetchAtBuffer_ = true;
-    }
-
     let currentMediaIndex = segmentInfo.mediaIndex;
 
     currentMediaIndex +=
       segmentInfo.playlist.mediaSequence - this.playlist_.mediaSequence;
 
+    if (!segmentInfo.isSyncRequest) {
```

Comment:
Can we actually get to this check if the segment is a sync request? In `handleSegment_` we return if it is a sync request and never append, so I would think this check is unnecessary.